### PR TITLE
Domain Layer Setup

### PR DESCRIPTION
**Pull request by Vanshbajaj**
1. Added the GraphQl
2. Domain Layer Setup -Added UseCase, Repository
3. Made Common Lint Adding in the Project.gradle

### CODE REVIEW

File: domain/src/main/java/com/practical/domain/repository/CharacterRepository.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.practical.domain.repository
+
+
+import com.domain.graphql.CharactersQuery
+import kotlinx.coroutines.flow.Flow
+
+interface CharacterRepository {
+    suspend fun getCharacters(): Flow<List<CharactersQuery.Data>>
```
Comment (shubhamsinghmutualmobile) [inline domain/src/main/java/com/practical/domain/repository/CharacterRepository.kt:8]: Are you going to write a subscription for getting `CharactersQuery.Data`? If not, consider using `Flow`s only inside `ViewModel`s.

File: domain/src/main/java/com/practical/domain/usecases/GetCharactersUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.practical.domain.usecases
+
+import com.domain.graphql.CharactersQuery
+import com.practical.domain.repository.CharacterRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+
+class GetCharactersUseCase @Inject constructor(private val repository: CharacterRepository) {
```
Comment (shubhamsinghmutualmobile) [inline domain/src/main/java/com/practical/domain/usecases/GetCharactersUseCase.kt:9]: Dependency injection components should not be a part of the `domain` layer. DI is a `presentation` layer concept.

**Review by shubhamsinghmutualmobile — CHANGES_REQUESTED** [review]

File: domain/build.gradle.kts [hunk 3]
```diff
@@ -43,11 +43,27 @@ android {
...
+        buildUponDefaultConfig = true
+    }
+
+    tasks.withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
+        reports {
+            html.enabled = true  // Enable HTML report
+            xml.enabled = false  // Disable XML report
+            txt.enabled = false  // Disable plain text report
+        }
+    }
```
Comment (shubhamsinghmutualmobile) [inline domain/build.gradle.kts:62]: You have written the same code thrice in 3 different modules. Can you try finding out a way to write this code only once and use it wherever you need?

Comment (shubhamsinghmutualmobile) [reply domain/build.gradle.kts:64]: Have you actually even resolved this comment?

**Review by shubhamsinghshubham777 — APPROVED** [review]

**Review by shubhamsinghmutualmobile — APPROVED** [review]

**Pull request merged**